File: src/chroma-shared/code/typescript/regex.ts
// @snippet:imports
import { DefaultEmbeddingFunction } from '@chroma-core/default-embed';
import { ChromaClient } from 'chromadb';

const client = new ChromaClient();
const embedder = new DefaultEmbeddingFunction();

const collection = await client.getOrCreateCollection({
  name: 'example-collection',
  embeddingFunction: embedder,
});
// @end

// @snippet:basic-regex
await collection.query({
  queryTexts: ['find support tickets with email addresses'],
  whereDocument: {
    $regex: '^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$',
  },
});
// @end

// @snippet:combined-filters
await collection.query({
  queryTexts: ['query1', 'query2'],
  whereDocument: {
    $and: [{ $contains: 'search_string_1' }, { $regex: '[a-z]+' }],
  },
});
// @end

File: src/chroma-shared/code/python/regex.py
# @snippet:imports
import chromadb

client = chromadb.HttpClient(host="localhost", port=8000)
collection = client.get_or_create_collection(name="my_collection")
# @end

# @snippet:basic-regex
results = collection.query(
    query_texts=["search query"],
    where_document={
        "$regex": "^tech.*"
    },
    n_results=10,
)
# @end

# @snippet:combined-filters
collection.query(
    query_texts=["query1", "query2"],
    where_document={
        "$and": [
            {"$contains": "search_string_1"},
            {"$regex": "[a-z]+"},
        ]
    },
)
# @end
